target_intacct/sinks.py

```diff
@@ -829,20 +832,44 @@ def purchase_orders_upload(self, record):
                 try:
                     item["projectid"] = self.projects[project_name]
                 except:
-                    return {
+                    return None, False, {
                         "error": f"ERROR: Project {project_name} does not exist. Did you mean any of these: {list(self.projects.keys())}?"
                     }
+
+            if item.get("projectid") and item.get("projectid") in self.projects_recordno.keys():
+                item["projectid"] = self.projects_recordno.get(item.get("projectid"))
+            elif item.get("projectid") and item.get("projectid") in self.projects.values():
+                item["projectid"] = self.projects.get(item.get("projectid"))
```

Comment:
there is an issue here, it should be
```suggestion
                item["projectid"] = item.get("projectid")
```